modules/host/contractmanager/sector.go

```diff
@@ -170,16 +174,32 @@ func (cm *ContractManager) HasSector(sectorRoot crypto.Hash) bool {
 	id := cm.managedSectorID(sectorRoot)
 
 	// Check if it exists
-	cm.wal.mu.Lock()
+	cm.sectorMu.Lock()
 	_, exists := cm.sectorLocations[id]
-	cm.wal.mu.Unlock()
+	cm.sectorMu.Unlock()
 
 	return exists
 }
 
+// lockSector grabs a sector lock.
+func (wal *writeAheadLog) lockSector(id sectorID) {
+	sl, exists := wal.cm.lockedSectors[id]
+	if exists {
+		sl.waiting++
+	} else {
+		sl = &sectorLock{
+			waiting: 1,
+		}
+		wal.cm.lockedSectors[id] = sl
+	}
+
+	// Block until the sector is available.
+	sl.mu.Lock()
+}
+
 // managedLockSector grabs a sector lock.
 func (wal *writeAheadLog) managedLockSector(id sectorID) {
```

Comment:
is this function still necessary? Should it call `lockSector`?

Reply:
changed it to call `lockSector`.